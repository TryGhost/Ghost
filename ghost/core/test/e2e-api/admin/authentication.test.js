const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');
const {anyErrorId} = matchers;

describe('Authentication API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('users');
        await agent.loginAsOwner();
    });

    describe('generateResetToken', function () {
        it('Cannot generate reset token without required info', async function () {
            await agent
                .post('authentication/password_reset')
                .expectStatus(400)
                .matchBodySnapshot({
                    errors: [{
                        id: anyErrorId
                    }]
                });
        });
    });
});
