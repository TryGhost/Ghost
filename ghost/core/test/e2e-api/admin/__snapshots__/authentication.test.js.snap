// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication API generateResetToken Cannot generate reset token without required info 1: [body] 1`] = `
Object {
  "errors": Array [
    Object {
      "code": null,
      "context": null,
      "details": null,
      "ghostErrorCode": null,
      "help": null,
      "id": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "message": "The server did not receive a valid email",
      "property": null,
      "type": "BadRequestError",
    },
  ],
}
`;
